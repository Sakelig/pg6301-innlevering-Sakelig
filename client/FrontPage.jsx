import { useLoader } from "./useLoader";
import { fetchJSON } from "./http";
import { Link } from "react-router-dom";
import * as React from "react";

export function FrontPage() {
  const {
    data: score,
    loading,
    reload,
  } = useLoader(async () => fetchJSON("/api/score"));

  return (
    <div>
      <h1>Quiz !</h1>
      {loading && <div>Loading ...</div>}
      {score && (
        <h3>
          You have answered {score.correct} of {score.answers} correctly so far!
        </h3>
      )}

      <ul>
        <li>
          <Link to={"/question"}>ny quiz</Link>
        </li>
      </ul>
    </div>
  );
}
