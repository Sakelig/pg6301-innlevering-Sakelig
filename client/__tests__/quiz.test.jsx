import * as React from "react";
import * as ReactDOM from "react-dom";
import { FrontPage } from "../FrontPage";
import { MemoryRouter } from "react-router-dom";

describe("Quiz Application", () => {
  it("show frontpage", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <FrontPage />
      </MemoryRouter>,
      element
    );
    //expect(element.querySelector("h1").innerHTML).toContain("Quiz 2 !");
    expect(element.innerHTML).toMatchSnapshot();
  });
});
