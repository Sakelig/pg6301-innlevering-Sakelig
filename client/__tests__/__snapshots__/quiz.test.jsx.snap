// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Application show frontpage 1`] = `"<div><h1>Quiz 2 !</h1><div>Loading ...</div><ul><li><a href=\\"/question\\">ny quiz</a></li></ul></div>"`;
